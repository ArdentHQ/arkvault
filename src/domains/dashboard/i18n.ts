export const translations = {
	DASHBOARD_PAGE: {
		TITLE: "Dashboard",
	},
	FILTER_WALLETS: {
		CRYPTOASSET: {
			DESCRIPTION: "Select the networks to show.",
			TITLE: "Filter Networks",
		},
		WALLETS: {
			DESCRIPTION: "Select which wallets to view.",
			TITLE: "Your Wallets",
		},
	},

	LATEST_TRANSACTIONS: {
		EMPTY_MESSAGE:
			"You don't have any transactions yet. Once transactions have been made they will show up in your history.",
		NO_RESULTS: "No <bold>{{type}}</bold> transactions could be found.",
		TITLE: "Latest Transactions",
	},

	PORTFOLIO_BREAKDOWN: {
		FILTERED: "Please enable at least one public network to display your portfolio report.",
	},

	PORTFOLIO_BREAKDOWN_DETAILS: {
		TITLE: "Portfolio Breakdown",
	},

	WALLET_CONTROLS: {
		CREATE: "Create",
		EMPTY_MESSAGE: "Your portfolio is currently empty. <1>Create</1> or <3>Import</3> a wallet to get started.",
		EMPTY_MESSAGE_FILTERED: "You don't have any wallets matching the selected networks.",
		EMPTY_MESSAGE_TYPE: "You don't have any <bold>{{type}}</bold> wallets.",
		EMPTY_MESSAGE_TYPE_FILTERED: "You don't have any <bold>{{type}}</bold> wallets matching the selected networks.",
		IMPORT: "Import",
		IMPORT_LEDGER: "Import Ledger",
	},
};
