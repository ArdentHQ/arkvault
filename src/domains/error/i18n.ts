export const translations = {
	APPLICATION: {
		DESCRIPTION: "It looks like something went wrong. Try visiting another page or reloading.",
		HELP_TEXT: "Please provide the error below when contacting support.",
		RELOAD: "Reload",
		TITLE: "Something went wrong",
	},
	OFFLINE: {
		DESCRIPTION: "It looks like there may be a problem connecting to the Internet. Check your connection.",
		TITLE: "Connectivity Issue",
	},
};
