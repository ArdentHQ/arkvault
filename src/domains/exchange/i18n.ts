export const translations = {
	EMPTY_MESSAGE: "You haven't made any swaps yet. Once you initiate a transaction they will show up here.",

	ERROR: {
		GENERIC: "Your order could not be placed",
		INVALID_ADDRESS: "The provided address is not a valid {{ticker}} address",
		INVALID_REFUND_ADDRESS: "The provided refund address is not a valid {{ticker}} address",
	},

	EXCHANGE_FORM: {
		AMOUNT_RECEIVED: "Amount Received",
		AMOUNT_SENT: "Amount Sent",
		CURRENCY_INPUT: "{{currency}} Input",
		CURRENCY_OUTPUT: "{{currency}} Output",
		ESTIMATED_ARRIVAL: "Estimated Arrival",
		ESTIMATED_RATE: "Estimated Rate",
		ESTIMATED_TIME: "â‰ˆ {{estimatedTime}} minutes",
		EXCHANGE_ADDRESS: "Exchange {{currency}} Address",
		FROM_CURRENCY: "From Currency",
		INPUT_TRANSACTION_ID: "Input Transaction ID",
		OUTPUT_TRANSACTION_ID: "Output Transaction ID",
		PAYIN_AMOUNT: "Payin Amount",
		PAYOUT_AMOUNT: "Payout Amount",
		RECIPIENT_PLACEHOLDER: "Enter the payout address",
		RECIPIENT_WALLET: "Recipient Wallet",
		REFUND_PLACEHOLDER: "Enter the refund address",
		REFUND_WALLET: "Refund Wallet",
		SUPPORT_INFO:
			"If you have any question about your exchange, please contact {{exchange}} support team via email at <linkEmail>{{email}}</linkEmail>",
		TERMS: "I've read and agree to the {{exchange}} <linkTerms>{{terms}}</linkTerms> and <linkPrivacyPolicy>{{privacy}}</linkPrivacyPolicy>",
		TO_CURRENCY: "To Currency",
		TRANSACTION_COMPLETE: "Transaction is Complete!",
		YOUR_ADDRESS: "Your {{currency}} Address",
		YOU_GET: "You Get",
		YOU_SEND: "You Send",
	},

	EXCHANGE_NAME: "Exchange Name",
	EXPECTED_AMOUNT_HINT: "The final payout amount might be different",

	MODAL_DELETE_EXCHANGE_TRANSACTION: {
		DESCRIPTION:
			"Are you sure you want to remove this swap from the transaction history? This action cannot be undone.",
		TITLE: "Remove Transaction",
	},

	NAVIGATION: {
		EXCHANGES: "Exchanges",
		TRANSACTIONS: "Transactions",
	},

	PAGE_EXCHANGES: {
		ADD_EXCHANGE: "Add Exchange",
		DELETE_CONFIRMATION: "The exchange transaction <bold>{{orderId}}</bold> has been deleted",
		EMPTY_MESSAGE: "There are currently no exchange providers available, please try again at a later time.",
		SUBTITLE: "Choose one of the exchange providers below to swap directly from within the wallet.",
		TITLE: "Exchange",
	},

	REFUND_WALLET: {
		ADD: "Add Refund Wallet",
		REMOVE: "Remove Refund Wallet",
	},

	SELECT_EXCHANGE_MESSAGE: "Select one of the exchanges you have installed",

	STATUS: {
		CONFIRMING: "Confirming",
		EXCHANGING: "Exchanging",
		EXPIRED: "Expired",
		FAILED: "Failed",
		FINISHED: "Finished",
		NEW: "New",
		REFUNDED: "Refunded",
		SENDING: "Sending",
		VERIFYING: "Verifying",
		WAITING: "Waiting",
	},

	TO_ADDRESS: "To Address",

	TRANSACTION_ID: "Transaction ID",
	TRANSACTION_STATUS: {
		AWAITING_DEPOSIT: "Awaiting Deposit",
		EXCHANGING: "Exchanging",
		EXPIRED: "This transaction has expired.",
		FAILED: "This transaction has failed.",
		REFUNDED: "This transaction has been refunded.",
		SENDING: "Sending",
		VERIFYING: "This transaction is being verified by the provider.",
	},

	VALIDATION: {
		INVALID_ADDRESS: "Not a valid {{ticker}} address",
		MIN_AMOUNT: "The minimal exchange amount is {{amount}} {{ticker}}",
		PAIR_NOT_AVAILABLE: "The pair <bold>{{from}}</bold> / <bold>{{to}}</bold> is not available.",
	},

	YOUR_EXCHANGE_LIST: "Your Exchange List",
};
