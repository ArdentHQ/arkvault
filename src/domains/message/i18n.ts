export const translations = {
  LEDGER_CONFIRMATION: {
    TITLE: "Confirm Your Signature",
  },

	PAGE_SIGN_MESSAGE: {
      AUTHENTICATION_STEP: {
      DESCRIPTION_ENCRYPTION_PASSWORD: "Enter your encryption password to sign the message.",
      DESCRIPTION_MNEMONIC: "Enter your mnemonic passphrase to sign the message.",
      DESCRIPTION_SECRET: "Enter your secret to sign the message.",
    },

		COPY_SIGNATURE: "Copy Signature",

		FORM_STEP: {
			DESCRIPTION_ENCRYPTION_PASSWORD: "Provide a message below and sign with your encryption password.",
			DESCRIPTION_LEDGER: "Provide a message below and sign with your ledger.",
			DESCRIPTION_MNEMONIC: "Provide a message below and sign with your mnemonic passphrase.",
			DESCRIPTION_SECRET: "Provide a message below and sign with your secret.",
			TITLE: "Sign Message",
		},

		SIGN: "Sign",

		SUCCESS_STEP: {
      TITLE: "Message Successfully Signed",
		},

		TITLE: "Sign Message",
	},

	PAGE_VERIFY_MESSAGE: {
		TITLE: "Verify Message",
	},

	SIGNATORY: "Signatory",
};
