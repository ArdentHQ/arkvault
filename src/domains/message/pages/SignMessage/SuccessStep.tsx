import { Services } from "@ardenthq/sdk";
import { Contracts as ProfileContracts } from "@ardenthq/sdk-profiles";
import React, { useRef } from "react";
import { useTranslation } from "react-i18next";

import { Address } from "@/app/components/Address";
import { FormField } from "@/app/components/Form";
import { StepHeader } from "@/app/components/StepHeader";
import { TextArea } from "@/app/components/TextArea";
import { DetailLabel, DetailTitle, DetailWrapper } from "@/app/components/DetailWrapper";
import { Icon } from "@/app/components/Icon";

export const SigningMessageInfo = ({
	message,
	wallet,
}: {
	message?: string;
	wallet: ProfileContracts.IReadWriteWallet;
}) => {
	const { t } = useTranslation();
	return (
		<div className="space-y-4">
			<DetailWrapper label={t("COMMON.SIGNING_WALLET")}>
				<div className="flex items-center justify-between space-x-2 sm:justify-start sm:space-x-0">
					<DetailTitle>{t("COMMON.ADDRESS")}</DetailTitle>
					<Address
						truncateOnTable
						address={wallet.address()}
						walletName={wallet.alias()}
						showCopyButton
						walletNameClass="text-theme-text text-sm leading-[17px] sm:leading-5 sm:text-base"
						addressClass="text-theme-secondary-500 dark:text-theme-secondary-700 text-sm leading-[17px] sm:leading-5 sm:text-base w-full w-3/4"
						wrapperClass="justify-end sm:justify-start"
					/>
				</div>
			</DetailWrapper>

			<DetailWrapper label={t("COMMON.MESSAGE")}>
				<p>{message}</p>
			</DetailWrapper>
		</div>
	);
};

export const SuccessStep = ({
	signedMessage,
	wallet,
}: {
	signedMessage: Services.SignedMessage;
	wallet: ProfileContracts.IReadWriteWallet;
}) => {
	const { t } = useTranslation();

	const messageReference = useRef();

	return (
		<section>
			<StepHeader
				title={t("MESSAGE.PAGE_SIGN_MESSAGE.SUCCESS_STEP.TITLE")}
				titleIcon={
					<Icon
						className="text-theme-success-100 dark:text-theme-success-900"
						dimensions={[24, 24]}
						name="Completed"
						data-testid="icon-Completed"
					/>
				}
			/>

			<div className="mt-4 space-y-4">
				<SigningMessageInfo message={signedMessage.message} wallet={wallet} />

				<div>
					<FormField name="json-signature">
						<DetailLabel>{t("MESSAGE.PAGE_SIGN_MESSAGE.FORM_STEP.SIGNATURE_JSON")}</DetailLabel>
						<TextArea
							className="mt-2 py-4"
							wrap="hard"
							ref={messageReference}
							defaultValue={JSON.stringify(signedMessage)}
							disabled
						/>
					</FormField>
				</div>
			</div>
		</section>
	);
};
