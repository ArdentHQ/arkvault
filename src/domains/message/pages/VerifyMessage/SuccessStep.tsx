import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";

import { useFormContext } from "react-hook-form";
import { FormField, FormLabel } from "@/app/components/Form";
import { StepHeader } from "@/app/components/StepHeader";
import { TextArea } from "@/app/components/TextArea";
import { Alert } from "@/app/components/Alert";
import { VerificationResult } from "@/domains/message/pages/VerifyMessage/VerifyMessage";
import { Icon } from "@/app/components/Icon";
import { Image } from "@/app/components/Image";

export const SuccessStep = ({ verificationResult }: { verificationResult?: VerificationResult }) => {
	const { t } = useTranslation();

	const isVerified = verificationResult?.verified;

	const { setError } = useFormContext();

	useEffect(() => {
		if (!isVerified) {
			setError("json-signature", { type: "manual" });
		}
	}, [isVerified, setError]);

	const getTitle = () => {
		if (isVerified) {
			return t("MESSAGE.PAGE_VERIFY_MESSAGE.SUCCESS_STEP.VERIFIED.TITLE");
		}

		return t("MESSAGE.PAGE_VERIFY_MESSAGE.SUCCESS_STEP.NOT_VERIFIED.TITLE");
	};

	const getDescription = () => {
		if (isVerified) {
			return t("MESSAGE.PAGE_VERIFY_MESSAGE.SUCCESS_STEP.VERIFIED.DESCRIPTION");
		}

		return t("MESSAGE.PAGE_VERIFY_MESSAGE.SUCCESS_STEP.NOT_VERIFIED.DESCRIPTION");
	};

	const titleIcon = () => {
		if (!isVerified) {
			return (
				<Image
					name="ErrorHeaderIcon"
					domain="transaction"
					className="block h-[22px] w-[22px]"
					useAccentColor={false}
				/>
			)
		}

		return (
			<Icon
				className="text-theme-success-100 dark:text-theme-success-900"
				dimensions={[24, 24]}
				name="Completed"
				data-testid="icon-Completed"
			/>
		)

	}

	return (
		<section>
			<StepHeader title={getTitle()}
				titleIcon={titleIcon()}
			/>

			<Alert className="mt-6 sm:mt-4" variant={isVerified ? "success" : "danger"}>
				{getDescription()}
			</Alert>

			<div className="pt-6 sm:pt-4">
				<FormField name="json-signature">
					<FormLabel label={t("MESSAGE.PAGE_VERIFY_MESSAGE.FORM_STEP.JSON_STRING")} />
					<TextArea
						className="py-4"
						wrap="hard"
						defaultValue={JSON.stringify({
							message: verificationResult?.message,
							signatory: verificationResult?.signatory,
							signature: verificationResult?.signature,
						})}
						disabled
					/>
				</FormField>
			</div>
		</section>
	);
};
