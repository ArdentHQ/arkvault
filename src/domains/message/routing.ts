import { ProfilePaths } from "@/router/paths";
import { RouteItem } from "@/router/router.types";
import preloadLazy from "@/utils/preload-lazy";

const SignMessage = preloadLazy(() => import("./pages/SignMessage"));
const VerifyMessage = preloadLazy(() => import("./pages/VerifyMessage"));

export const MessageRoutes: RouteItem[] = [
	{
		component: SignMessage,
		exact: true,
		path: ProfilePaths.SignMessageWallet,
	},
	{
		component: SignMessage,
		exact: true,
		path: ProfilePaths.SignMessage,
	},
	{
		component: VerifyMessage,
		exact: true,
		path: ProfilePaths.VerifyMessage,
	},
	{
		component: VerifyMessage,
		exact: true,
		path: ProfilePaths.VerifyMessageWallet,
	},
];
