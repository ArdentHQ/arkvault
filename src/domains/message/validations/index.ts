export * from "./SignMessage";
export * from "./VerifyMessage";
