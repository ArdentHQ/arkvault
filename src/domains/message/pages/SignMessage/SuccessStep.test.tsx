/* eslint-disable @typescript-eslint/require-await */
import { createHashHistory } from "history";
import React from "react";
import { Route } from "react-router-dom";

import { translations as messageTranslations } from "@/domains/message/i18n";
import { env, getDefaultProfileId, renderResponsiveWithRoute, screen, waitFor } from "@/utils/testing-library";
import { SuccessStep } from "./SuccessStep";

const history = createHashHistory();

describe("SignMessage success step", () => {
	it.each(["xs", "lg"])("should render success step in %s", async (breakpoint) => {
		const profile = env.profiles().findById(getDefaultProfileId());

		const { asFragment } = renderResponsiveWithRoute(
			<Route path="/profiles/:profileId/wallets/:walletId/sign-message">
				<SuccessStep
					wallet={profile.wallets().first()}
					signedMessage={{ message: "Test Message", signatory: "!", signature: "1", mnemonic: "1" }}
				/>
			</Route>,
			breakpoint,
			{
				history,
				route: `/profiles/${getDefaultProfileId()}/wallets/${profile.wallets().first().id()}/sign-message`,
			},
		);

		await waitFor(() => {
			expect(
				screen.findByRole("heading", { name: messageTranslations.PAGE_SIGN_MESSAGE.FORM_STEP.TITLE }),
			).resolves.toBeDefined();
		});

		expect(screen.getByText("Test Message")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});
});
