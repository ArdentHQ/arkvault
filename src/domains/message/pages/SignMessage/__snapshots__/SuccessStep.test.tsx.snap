// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignMessage success step > should render success step in lg 1`] = `
<DocumentFragment>
  <section>
    <div
      class="flex items-end justify-between bg-theme-background"
    >
      <div
        class="space-y-2"
      >
        <div
          class="flex items-center gap-3"
        >
          <h1
            class="mb-0 text-2xl leading-[29px]"
            data-testid="header__title"
          >
            Message Successfully Signed
          </h1>
        </div>
      </div>
    </div>
    <div
      class="no-ligatures mt-4 md:mt-2 css-1slril7"
      data-testid="TransactionDetail"
    >
      <div
        class="w-40 flex-1 space-y-2 whitespace-nowrap"
        data-testid="TransactionDetail--desktop"
      >
        <div
          class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Signatory
        </div>
        <div
          class="flex items-center font-semibold"
        >
          <div
            class="w-0 flex-1 text-right md:text-left"
          >
            <div
              class="flex overflow-hidden whitespace-nowrap items-center space-x-2 w-full"
            >
              <span
                class="font-semibold text-base text-theme-text"
                data-testid="Address__alias"
              >
                <span
                  class="no-ligatures css-9nx3be"
                  data-testid="TruncateEnd"
                >
                  ARK Wallet 1
                </span>
              </span>
              <span
                class="no-ligatures min-w-0 overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                data-testid="Address__address"
              >
                D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="shrink-0 css-1n9jllj"
        data-testid="Avatar"
      >
        <div
          class="inline-flex h-full w-full items-center justify-center overflow-hidden align-middle rounded-full"
        >
          <img
            alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
            src="data:image/svg+xml;utf8,<svg version="1.1" xmlns="http://www.w3.org/2000/svg" class="picasso" width="100" height="100" viewBox="0 0 100 100"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill="rgb(0, 150, 136)" width="100" height="100"/><circle r="50" cx="80" cy="50" fill="rgb(76, 175, 80)"/><circle r="35" cx="100" cy="40" fill="rgb(156, 39, 176)"/><circle r="40" cx="90" cy="70" fill="rgb(3, 169, 244)"/></svg>"
            title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
          />
        </div>
      </div>
    </div>
    <div
      class="no-ligatures css-1slril7"
      data-testid="TransactionDetail"
    >
      <div
        class="w-40 flex-1 space-y-2 whitespace-nowrap"
        data-testid="TransactionDetail--desktop"
      >
        <div
          class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Message
        </div>
        <div
          class="flex items-center font-semibold"
        >
          <span
            class="min-w-0 whitespace-normal break-words text-right md:text-left"
          >
            Test Message
          </span>
        </div>
      </div>
    </div>
    <div
      class="pt-4 md:pt-6"
    >
      <fieldset
        class="flex min-w-0 flex-col css-1bcidjh"
      >
        <label
          class="FormLabel mb-2 flex text-sm font-semibold leading-[17px] text-theme-secondary-text transition-colors duration-100"
          data-testid="FormLabel"
          for="json-signature"
        >
          JSON String
        </label>
        <div
          class="relative"
        >
          <div
            class="pointer-events-none absolute bottom-4 right-3 z-10 hidden h-0 w-0 -rotate-45 border-x-8 border-t-8 border-x-transparent border-t-theme-secondary-400 dark:border-t-theme-secondary-700 md:block"
          />
          <div
            class="py-4 css-1prn1rr"
            disabled=""
          >
            <div
              class="relative flex h-full flex-1"
            >
              <textarea
                autocomplete="off"
                class="no-ligatures w-full border-none resize-none min-h-[8rem] md:min-h-auto md:resize-y resize-none text-theme-secondary-text css-1vs1awz"
                data-testid="TextArea"
                disabled=""
                name="json-signature"
                style="height: 4px;"
                type="text"
                wrap="hard"
              >
                {"message":"Test Message","mnemonic":"1","signatory":"!","signature":"1"}
              </textarea>
            </div>
          </div>
        </div>
      </fieldset>
    </div>
  </section>
</DocumentFragment>
`;

exports[`SignMessage success step > should render success step in xs 1`] = `
<DocumentFragment>
  <section>
    <div
      class="flex items-end justify-between bg-theme-background"
    >
      <div
        class="space-y-2"
      >
        <div
          class="flex items-center gap-3"
        >
          <h1
            class="mb-0 text-2xl leading-[29px]"
            data-testid="header__title"
          >
            Message Successfully Signed
          </h1>
        </div>
      </div>
    </div>
    <div
      class="no-ligatures mt-4 md:mt-2 css-1slril7"
      data-testid="TransactionDetail"
    >
      <div
        class="items-top flex w-full justify-between space-x-4 md:items-center"
        data-testid="TransactionDetail--mobile"
      >
        <div
          class="text-md md:theme-text group relative m-0 select-none border-theme-secondary-300 text-left font-semibold text-theme-secondary-700 first:pl-0 last:pr-0 dark:border-theme-secondary-800 dark:text-theme-secondary-500"
        >
          Signatory
        </div>
        <div
          class="flex min-w-0 grow items-center justify-end space-x-4"
        >
          <div
            class="md:theme-text flex w-full justify-end text-theme-secondary-700 dark:text-theme-secondary-500"
          >
            <div
              class="w-0 flex-1 text-right md:text-left"
            >
              <div
                class="flex overflow-hidden whitespace-nowrap items-center space-x-2 w-full"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="no-ligatures css-9nx3be"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="no-ligatures min-w-0 overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
          <div
            class="shrink-0 css-1yk316i"
            data-testid="Avatar"
          >
            <div
              class="inline-flex h-full w-full items-center justify-center overflow-hidden align-middle rounded-full"
            >
              <img
                alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                src="data:image/svg+xml;utf8,<svg version="1.1" xmlns="http://www.w3.org/2000/svg" class="picasso" width="100" height="100" viewBox="0 0 100 100"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill="rgb(0, 150, 136)" width="100" height="100"/><circle r="50" cx="80" cy="50" fill="rgb(76, 175, 80)"/><circle r="35" cx="100" cy="40" fill="rgb(156, 39, 176)"/><circle r="40" cx="90" cy="70" fill="rgb(3, 169, 244)"/></svg>"
                title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="no-ligatures css-1slril7"
      data-testid="TransactionDetail"
    >
      <div
        class="items-top flex w-full justify-between space-x-4 md:items-center"
        data-testid="TransactionDetail--mobile"
      >
        <div
          class="text-md md:theme-text group relative m-0 select-none border-theme-secondary-300 text-left font-semibold text-theme-secondary-700 first:pl-0 last:pr-0 dark:border-theme-secondary-800 dark:text-theme-secondary-500"
        >
          Message
        </div>
        <div
          class="flex min-w-0 grow items-center justify-end space-x-4"
        >
          <div
            class="md:theme-text flex w-full justify-end text-theme-secondary-700 dark:text-theme-secondary-500"
          >
            <span
              class="min-w-0 whitespace-normal break-words text-right md:text-left"
            >
              Test Message
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pt-4 md:pt-6"
    >
      <fieldset
        class="flex min-w-0 flex-col css-1bcidjh"
      >
        <label
          class="FormLabel mb-2 flex text-sm font-semibold leading-[17px] text-theme-secondary-text transition-colors duration-100"
          data-testid="FormLabel"
          for="json-signature"
        >
          JSON String
        </label>
        <div
          class="relative"
        >
          <div
            class="pointer-events-none absolute bottom-4 right-3 z-10 hidden h-0 w-0 -rotate-45 border-x-8 border-t-8 border-x-transparent border-t-theme-secondary-400 dark:border-t-theme-secondary-700 md:block"
          />
          <div
            class="py-4 css-1prn1rr"
            disabled=""
          >
            <div
              class="relative flex h-full flex-1"
            >
              <textarea
                autocomplete="off"
                class="no-ligatures w-full border-none resize-none min-h-[8rem] md:min-h-auto md:resize-y resize-none text-theme-secondary-text css-1vs1awz"
                data-testid="TextArea"
                disabled=""
                name="json-signature"
                style="height: 4px;"
                type="text"
                wrap="hard"
              >
                {"message":"Test Message","mnemonic":"1","signatory":"!","signature":"1"}
              </textarea>
            </div>
          </div>
        </div>
      </fieldset>
    </div>
  </section>
</DocumentFragment>
`;
